### PR TITLE
Add GraphHopper Attribution

### LINKED ISSUE #3: Extend Attribution List
Add GraphHopper to the attribution list.

### PR DESCRIPTION
**Pull request by boldtrn**
This PR adds attribution for GraphHopper to the Map. Fixes #3.

Unfortunately the Mapbox SDK makes this rather hard. That's why we had to extend the AttributionDialogManager. I wasn't able to register GH as a Source in the map.

### CODE REVIEW

File: app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        private final MapboxMap mapboxMap;
+
+        AttributionBuilder(MapboxMap mapboxMap) {
+            this.mapboxMap = mapboxMap;
+        }
+
+        private Set<Attribution> build() {
+            List<String> attributions = new ArrayList<>();
+            attributions.add("<a href=\"https://www.graphhopper.com/\" target=\"_blank\">&copy; GraphHopper</a>");
```
Comment (karussell) [inline app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: Here I would like to use "GraphHopper API" like we defined https://graphhopper.com/api/1/docs/attribution/

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<resources>
-    <string name="app_name">GH Nav Sample</string>
+<resources xmlns:tools="http://schemas.android.com/tools">
+    <string name="app_name" tools:override="true">GH Navigation Example</string>
```
Comment (karussell) [inline app/src/main/res/values/strings.xml]: Maybe just "GraphHopper Demo"?

File: app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            for (Source source : mapboxMap.getSources()) {
+                attributions.add(source.getAttribution());
+            }
+
+            return new AttributionParser.Options()
+                    .withCopyrightSign(true)
+                    // TODO when using Mapbox as Tilesource we should keep this, should we automatically remove it otherwise?
+                    .withImproveMap(true)
+                    .withTelemetryAttribution(true)
```
Comment (karussell) [inline app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: Shouldn't this be false per default as we'll soon switch to MapTiler` or did I misunderstood the option somehow?

**Review by karussell — APPROVED** [review]
Haven't run the app but looks good.

File: app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java (hunk 1 above)
Comment (boldtrn) [reply app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: Ok, I will replace it with "Powered by GraphHopper API"?

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (boldtrn) [reply app/src/main/res/values/strings.xml]: Sure, I can change that

File: app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java (hunk 1 above)
Comment (karussell) [reply app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: Or `&copy; GraphHopper API` is sufficient as we can assume a small screen for now :)

File: app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java (hunk 3 above)
Comment (boldtrn) [reply app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: I had the same thought. Currently, the default for the app is still Mapbox, so I kept it for now.

If we want to make MapTiller or any other Tile provider the default, we can remove this or maybe even better, make this configureable in the dev config in the future?

Comment (boldtrn) [reply app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: I will also disable Mapbox Telemetry by default, so we can remove the Telemetry Attribution.

Comment (karussell) [reply app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: Yes, lets disable this and disable mapbox attribution when we switch to MapTiler (not necessary to make it configurable at the moment)

**Comment by boldtrn** [comment]
Just to double check, should we merge the current state?

File: app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java (hunk 3 above)
Comment (boldtrn) [reply app/src/main/java/graphhopper/com/graphhopper_navigation_example/GHAttributionDialogManager.java]: > not necessary to make it configurable at the moment

This is actually pretty easy, so I think we could think about this

**Comment by karussell** [comment]
Sure

**Comment by boldtrn** [comment]
Ok, I just found a minor issue. Will update this PR soon and merge afterwards :+1:

**Comment by karussell** [comment]
Thanks!

**Pull request merged**